kyaml/kio/kioutil/kioutil_test.go

```diff
@@ -75,3 +76,257 @@ y: z
 
 	assert.Equal(t, strings.TrimSpace(input), strings.TrimSpace(actual.String()))
 }
+
+func TestDefaultPathAnnotation(t *testing.T) {
+	var tests = []struct {
+		dir      string
+		input    string // input
+		expected string // expected result
+		name     string
+	}{
+		{
+			`foo`,
+			`apiVersion: v1
+kind: Bar
+metadata:
+  name: a
+  namespace: b
+`,
+			`apiVersion: v1
+kind: Bar
+metadata:
+  name: a
+  namespace: b
+  annotations:
+    config.kubernetes.io/path: 'foo/b/bar_a.yaml'
+`, `with namespace`},
+		{
+			`foo`,
+			`apiVersion: v1
+kind: Bar
+metadata:
+  name: a
+`,
+			`apiVersion: v1
+kind: Bar
+metadata:
+  name: a
+  annotations:
+    config.kubernetes.io/path: 'foo/bar_a.yaml'
```

Comment:
Pros/cons of using 'default' for Resource in default namespace?

dir/default/pod_foo.yaml?

Also, how are you handling cluster-scoped resources:

dir/cluster-rolebinding_bar.yaml ?

Reply:
> Pros/cons of using 'default' for Resource in default namespace?
> dir/default/pod_foo.yaml?

not specifying a namespace doesn't mean it will end up in default, the namespace may be specified as a flag in kubectl

> Also, how are you handling cluster-scoped resources:
> dir/cluster-rolebinding_bar.yaml ?

Yes, these don't have a namespace